app/services/notifications_cleanup_service.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class NotificationsCleanupService
+  # This can be expensive, so instead of cleaning up everything
+  # at once, we limit the number of accounts per run and run
+  # this more often.
+  ACCOUNTS_PER_RUN = 5
+
+  # Unfiltered notifications do not need any additional
+  # processing and can be deleted via SQL. This means we
+  # can safely delete a large number in one run.
+  UNFILTERED_DELETES_PER_ACCOUNT = 100_000
```

Comment:
Deleting via SQL `DELETE` is quite quick, so a larger value here might be possible. *But* while pg returns quickly from `DELETE` commands, the real work seem to start after that. I noticed very high CPU usage of pg after I deleted a large number of entries. So maybe it is best to not set this too high.